src/util/system.cpp

```diff
@@ -945,7 +945,7 @@ std::string ArgsManager::GetChainName() const
     const bool is_chain_arg_set = IsArgSet("-chain");
 
     if ((int)is_chain_arg_set + (int)fRegTest + (int)fSigNet + (int)fTestNet > 1) {
-        throw std::runtime_error("Invalid combination of -regtest, -signet, -testnet and -chain. Can use at most one.");
+        return "Invalid combination of -regtest, -signet, -testnet or -chain";
```

Comment:
I am not convinced that an error should be encoded as special chain name of "Invalid...". Even if this compiles right now, this is asking for issues down the road where the caller doesn't sanitize the chain name.

Reply:
This is what std::optional is for! Let's use it.

Reply:
Doing this with std::optional everywhere seems pretty ugly, here's an alternative approach: https://github.com/ajtowns/bitcoin/commits/202102-getchainname . Probably better to have `string GetChainName()` and `string GetChainNameCheck()` and a private `bool GetChainName(string&)` that they both call than moving the exception message to the various callers though.